examples/search_augmentation.ipynb

```diff
@@ -0,0 +1,472 @@
+    "sorted_articles = sorted(scored_articles, key=lambda x: x[1], reverse=True)\n",
+    "\n",
+    "# Print top 5 articles\n",
+    "print(\"Top 5 articles:\", \"\\n\")\n",
+    "\n",
+    "for article, score in sorted_articles[0:5]:\n",
+    "    print(\"Title:\", article[\"title\"])\n",
+    "    print(\"Description:\", article[\"description\"])\n",
+    "    print(\"Content:\", article[\"content\"][0:100] + \"...\")\n",
+    "    print(\"Score:\", score)\n",
+    "    print()\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Awesome! These results look a lot more relevant to our original query. Now, let's use the top 20 results to generate a final answer.\n",
+    "\n",
+    "## 3. Answer\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "Recently, several events have taken place in San Francisco. The city's mayor, London Breed, stated the need for action on the drug crisis, even if services aren't accepted [ABC7 News Bay Area](https://consent.google.com/ml?continue=https://news.google.com/rss/articles/CCAiC1pHdmR6QkQ0ZnBZmAEB?oc%3D5&gl=FR&hl=en-US&cm=2&pc=n&src=1). A few mass shootings have occurred in San Francisco's Mission District, with one resulting in 9 injuries [KTVU FOX 2 San Francisco](https://consent.google.com/ml?continue=https://news.google.com/rss/articles/CCAiC1NhNDdrRE9GQUs4mAEB?oc%3D5&gl=FR&hl=en-US&cm=2&pc=n&src=1). The Park Hotels company has abandoned 9% of San Francisco's hotel rooms due to weak demand in the city [Skift](https://skift.com/2023/06/08/park-hotels-gives-up-on-san-franciscos-doggedly-weak-demand/). In response to the drug crisis, the San Francisco Sheriff has deployed deputies to arrest drug dealers [KTVU FOX 2 San Francisco](https://consent.google.com/ml?continue=https://news.google.com/rss/articles/CCAiC0VvcXlvaUlZSng4mAEB?oc%3D5&gl=FR&hl=en-US&cm=2&pc=n&src=1). A gas leak and water main break led to evacuations in the city [KTVU FOX 2 San Francisco](https://consent.google.com/ml?continue=https://news.google.com/rss/articles/CCAiC1M2UXotMU1qRTVZmAEB?oc%3D5&gl=FR&hl=en-US&cm=2&pc=n&src=1). Additionally, Hilton SF Union Square and Parc 55's owner has stopped payments on loans for these properties [ABC7 News Bay Area](https://consent.google.com/ml?continue=https://news.google.com/rss/articles/CCAiC3dpQk55U3I0SHBNmAEB?oc%3D5&gl=FR&hl=en-US&cm=2&pc=n&src=1)."
+      ],
+      "text/plain": [
+       "<IPython.core.display.Markdown object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "formatted_top_results = [\n",
+    "    {\n",
+    "        \"title\": article[\"title\"],\n",
+    "        \"description\": article[\"description\"],\n",
+    "        \"url\": article[\"url\"],\n",
+    "    }\n",
+    "    for article, _score in sorted_articles[0:50]\n",
```

Comment:
could be interesting to do the top 5 instead of top 50 so that the user can personally judge the model (otherwise they're just taking this answer on faith). could also put the raw and re-ranked answers side-by-side to see the improvement caused by re-ranking.